tests/containers/apparmor.pm

```diff
@@ -0,0 +1,57 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Package: apparmor-utils
+# Summary: Test apparmor utilities
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use services::apparmor;
+use utils;
+
+sub is_enabled {
+    # according to Linux' Documentation/admin-guide/LSM/index.rst,
+    # the current security modules can be found in /sys/kernel/security/lsm
+    return script_run 'grep apparmor /sys/kernel/security/lsm' == 0;
+}
+
+sub assert_running {
+    services::apparmor::check_service();
+    services::apparmor::check_aa_status();
+}
+
+sub assert_functional {
+    my $dnsmasq_file = '/etc/apparmor.d/usr.sbin.dnsmasq';
+    if (script_run "test -f $dnsmasq_file") {
+        record_info('Skipped', "File $dnsmasq_file not found, skipping assert_functional test...");
+        return 1;
+    }
+
+    script_run "cp $dnsmasq_file /tmp/usr.sbin.dnsmasq";
+    script_run "echo this should fail > $dnsmasq_file";
+
+    my $ret = systemctl('restart apparmor', ignore_failure => 1);
+    if ($ret == 0) {
+        die 'AppArmor should not be able to restart with an erroneous profile'
+    }
+
+    # restore the backed up profile
+    script_run "mv /tmp/usr.sbin.dnsmasq $dnsmasq_file";
+    systemctl('restart apparmor');
+}
+
+sub run {
+    select_console 'root-console';
+
+    if (not is_enabled) {
```

Comment:
looks nice, but we tend to use `unless (is_enabled)` in such a case. But that is just a neat pick, feel free to ignore

Reply:
Fixed in [398823](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/3988234a5118e4f6ef36386634d858b9d1c148a5).